### PR TITLE
JS: Avoid using jquery object multiple times, extract as variable #7987

### LINKED ISSUE #7987: JS: Avoid using jquery object multiple times, extract as variable
when using a jQuery object multiple times `(as is the case here for $(entry))`, it is preferable to extract the jQuery object as a variable first (prefixing the variable name with a $ to indicate it is a jQuery object), and then reuse that variable in later calls eg:

**Expected:**
```javascript
         const $entry = $(e.target); 
         const courseId = $entry.data('courseid'); 
         const studentEmail = $entry.data('studentemail'); 
         const googleId = $entry.data('googleid'); 
```
**Current:**
https://github.com/TEAMMATES/teammates/blob/cb4f03f4071ac4f37e5a1d683e5587508f0a622c/src/main/webapp/dev/js/main/adminSearch.es6#L125-L128

The main difference between declaring `entry = e.target`, followed by using `$(entry)` later, and declaring `$entry = $(e.target)` and reusing `$entry` would be efficiency (no need to create the same object multiple times in the latter case).

### PR DESCRIPTION
**Pull request by AlexM9200**
Fixes #7987 

**PR Checklist**

**Outline of Solution**

Extracted jQuery object as a variable as outlined by the issue

### CODE REVIEW

**Comment by AlexM9200** [comment]
Ready for review

File: src/main/webapp/dev/js/main/instructorStudentList.es6 [hunk 1]
```diff
@@ -96,9 +96,10 @@ function filterSection() {
  */
 function filterTeam() {
     $('input[id^="team_check"]').each(function () {
-        const courseIdx = $(this).attr('id').split('-')[1];
-        const sectionIdx = $(this).attr('id').split('-')[2];
-        const teamIdx = $(this).attr('id').split('-')[3];
+        const $this = $(this);
```
Comment (wkurniawan07) [inline src/main/webapp/dev/js/main/instructorStudentList.es6]: It looks like there is no other usage of `$(this)` other than the next three lines. You can make `$(this).attr('id').split('-')` the variable instead.

File: src/main/webapp/dev/js/main/instructorStudentList.es6 [hunk 2]
```diff
@@ -96,9 +96,10 @@ function filterSection() {
  */
 function filterTeam() {
     $('input[id^="team_check"]').each(function () {
-        const courseIdx = $(this).attr('id').split('-')[1];
-        const sectionIdx = $(this).attr('id').split('-')[2];
-        const teamIdx = $(this).attr('id').split('-')[3];
+        const $thisAttr = $(this).attr('id').split('-');
```
Comment (wkurniawan07) [inline src/main/webapp/dev/js/main/instructorStudentList.es6]: This is a generic array and no longer a jQuery object, as such do not prefix it with an `$`.
Also, the variable name could be improved. `thisAttr` is too generic; what kind of `attr` (it is very specific here)? In addition, the variable name should suggest that it is an array.

Comment (AlexM9200) [reply src/main/webapp/dev/js/main/instructorStudentList.es6]: On second though teamAttributes is wrong. Would `indexArray` be better?

**Pull request merged**